tests/integration/sync_test.go

```diff
@@ -8052,3 +8052,154 @@ func Test_Sync_Plugins_Nested_Foreign_Keys_EE_3x(t *testing.T) {
 		})
 	}
 }
+
+// test scope:
+//
+// - >=3.0.0
+// - konnect
+func Test_Sync_Scoped_Plugins_Check_Conflicts_3x(t *testing.T) {
+	runWhenKongOrKonnect(t, ">=3.0.0")
+	setup(t)
+
+	client, err := getTestClient()
+	require.NoError(t, err)
+	ctx := t.Context()
+
+	stateFiles := []string{
+		"testdata/sync/036-scoped-plugins-validation/3x/plugins1.yaml",
+		"testdata/sync/036-scoped-plugins-validation/3x/plugins2.yaml",
+	}
+
+	tests := []struct {
+		name       string
+		multiSync  bool
+		emptyState bool
+	}{
+		{
+			name:       "sync plugin files one by one, existing state: empty",
+			multiSync:  false,
+			emptyState: true,
+		},
+		{
+			name:       "sync plugin files one by one, existing state: non-empty",
+			multiSync:  false,
+			emptyState: false,
+		},
+		{
+			name:       "sync plugin files together, existing state: empty",
+			multiSync:  true,
+			emptyState: true,
+		},
+		{
+			name:       "sync plugin files together, existing state: non-empty",
+			multiSync:  true,
+			emptyState: false,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			mustResetKongState(ctx, t, client, deckDump.Config{})
+			if !tc.emptyState {
+				// syncing first state file to create initial state
+				err := sync(stateFiles[0])
+				require.NoError(t, err)
+			}
+
+			if tc.multiSync {
+				err := multiFileSync(stateFiles)
+				require.NoError(t, err)
+
+				// re-sync with no error
+				err = multiFileSync(stateFiles)
+				require.NoError(t, err)
+
+				return
+			}
+
+			for _, s := range stateFiles {
+				err := sync(s)
+				require.NoError(t, err)
+
+				// re-sync with no error
+				err = sync(s)
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+// test scope:
+//
+// - >=2.8.0 <3.0.0
+func Test_Sync_Scoped_Plugins_Check_Conflicts(t *testing.T) {
+	runWhenKongOrKonnect(t, ">=2.8.0 <3.0.0")
+	setup(t)
+
+	client, err := getTestClient()
+	require.NoError(t, err)
+	ctx := t.Context()
+
+	stateFiles := []string{
+		"testdata/sync/036-scoped-plugins-validation/3x/plugins1.yaml",
```

Comment:
I see the files we are using are same for tests scoped `>=2.8.0 <3.0.0` and `>=3.0.0` - why do we need different tests?

Reply:
The v1 files are unused?
tests/integration/testdata/sync/036-scoped-plugins-validation/plugins1.yaml